test/test.py

```diff
@@ -0,0 +1,305 @@
+import unittest
+from pathlib import Path
+from rdflib.compare import isomorphic
+from src.owl2Shacl import create_shacl, rdf_validate
+from rdflib import Graph, SH, BNode, RDF, XSD, Literal, Namespace, OWL, RDFS
+from rdflib.term import URIRef
+
+IES = Namespace("http://ies.data.gov.uk/ontology/ies4#")
+DATA = Namespace("http://data.gov.uk/testdata#")
+ISO8601 = Namespace("http://iso.org/iso8601#")
+SPARX = Namespace("http://data.sparxsystems.com#")
+SH_CLASS = URIRef("http://www.w3.org/ns/shacl#class")
+SH_OR = URIRef("http://www.w3.org/ns/shacl#or")
+
+
+class TestOwl2Shacl(unittest.TestCase):
+
+    # @unittest.skip("done")
+    def test_valid_data(self):
+        expected_result = Graph()
+        vrep = BNode()
+        expected_result.add((vrep, RDF.type, SH.ValidationReport))
+        expected_result.add((vrep, SH.conforms, Literal("true", datatype=XSD.boolean)))
+        # ont_file = Path("IES Specification Docs/Information Exchange Standard r4.2.0.n3")
+        ont_file = Path("IES Specification Docs/IES4.ttl")
+        ont_graph, sh_graph = create_shacl(ont_file)
+        conforms, results_graph, results_text = rdf_validate(
+            "../Sample Data/hospital.ttl", ont_graph, sh_graph)
+        self.assertTrue(conforms)
+        self.assertEquals("Validation Report\nConforms: True\n", results_text)
+        assert isomorphic(expected_result, results_graph)
+
+    # @unittest.skip("done")
+    def test_object_property_not_class(self):
+        expected_result = Graph()
+        vrep = BNode()
+        vres = BNode()
+        expected_result.add((vrep, RDF.type, SH.ValidationReport))
```

Comment:
probably clearer to write the expected in ttl as a string, then parse into a graph

Reply:
Probably, I have used a range of techniques to create the expected results as I tried out different things